src/domain/eventSearch/utils.tsx

```diff
@@ -215,10 +228,14 @@ export const getEventSearchVariables = ({
     keywordAnd.push('yso:p4354');
   }
 
+  const keywords =
+    MAPPED_CATEGORY_TERMS[searchSource || EVENT_SEARCH_SOURCES.EVENTS];
   const mappedCategories: string[] = categories
-    .map((category) => MAPPED_CATEGORIES[category])
+    .map(
+      (category) =>
+        MAPPED_CATEGORIES[searchSource || EVENT_SEARCH_SOURCES.EVENTS][category]
+    )
```

Comment:
at line 179 you could add default value to searchSource:

`searchSource = EVENT_SEARCH_SOURCES.EVENTS,`

Reply:
Then there would be no need for || which makes this little bit hard to read

Reply:
it depends if we use eent source or they will create the keyword set in courses. i would still want to see api

Reply:
But you are using `EVENT_SEARCH_SOURCES.EVENTS` as a default value anyway atm for `MAPPED_CATEGORY_TERMS` and `MAPPED_CATEGORIES`. So for now at least think it would make more sense to have the default value set to searchSource. If i'm understading this correctly

Reply:
of course thing can change and event search also needs to use keywordSet but then this has to be thought again. Or if you have some idea how we could handle this now to have it work with keywordSet from event search also

Reply:
yes, i am using it as default as its optional param, and undefined cannot be used as index. this method is used a lot by event search, so when no search source, then we use EVENT_SEARCH_SOURCES.EVENTS (and as default), when we define search source, then we use different. Now i defined 2 sources, events and courses, which means basically that we query linked_events or linked_courses. I still think that they will create the keyword_set for courses under the linked_courses, if not, i can fix that later when implementing the rest.